Legobot/Connectors/Discord.py

```diff
@@ -0,0 +1,318 @@
+from websocket import create_connection
+import json
+import logging
+import requests
+import sys
+import threading
+import time
+
+from Legobot.Lego import Lego
+from Legobot.Message import Message, Metadata
+from Legobot.Utilities import Utilities
+
+__author__ = "Bren Briggs (fraq)"
+__copyright__ = "Copyright 2017, Legobot"
+
+logger = logging.getLogger(__name__)
+
+
+class Heartbeat(threading.Thread, object):
+    """
+    Manage heartbeat connection to Discord
+
+    Attributes:
+        ws: Websocket connection to Discord
+        interval (float): time between heartbeat pings to the server
+
+    """
+
+    def __init__(self, ws, interval):
+        self.ws = ws
+        self.interval = float(interval) / 1000.0
+        self.sequence = 0
+        logger.debug("Heartbeat handler initialized \
+                     with interval of {}".format(self.interval))
+        threading.Thread.__init__(self)
+
+    def send(self, ws, seq):
+        """
+        Sends heartbeat message to Discord
+
+        Attributes:
+            ws: Websocket connection to discord
+            seq: Sequence number of heartbeat
+        """
+
+        payload = {'op': 1, 'd': seq}
```

Comment:
I don't understand this payload - it seems to have a magic number in it and the terms aren't clear. I'm guessing this is a format required by Discord, but a comment might be helpful here.